x-pack/plugins/infra/server/routes/log_analysis/results/log_entry_categories.ts

```diff
@@ -5,36 +5,29 @@
  */
 
 import Boom from 'boom';
-
-import { pipe } from 'fp-ts/lib/pipeable';
-import { fold } from 'fp-ts/lib/Either';
-import { identity } from 'fp-ts/lib/function';
-import { schema } from '@kbn/config-schema';
-import { InfraBackendLibs } from '../../../lib/infra_types';
 import {
-  LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
   getLogEntryCategoriesRequestPayloadRT,
   getLogEntryCategoriesSuccessReponsePayloadRT,
+  LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
 } from '../../../../common/http_api/log_analysis';
-import { throwErrors } from '../../../../common/runtime_types';
-import { NoLogAnalysisResultsIndexError } from '../../../lib/log_analysis';
-
-const anyObject = schema.object({}, { unknowns: 'allow' });
+import { createValidationFunction } from '../../../../common/runtime_types';
+import type { InfraBackendLibs } from '../../../lib/infra_types';
+import {
+  getTopLogEntryCategories,
+  NoLogAnalysisResultsIndexError,
+} from '../../../lib/log_analysis';
+import { assertHasInfraMlPlugins } from '../../../utils/request_context';
 
-export const initGetLogEntryCategoriesRoute = ({
-  framework,
-  logEntryCategoriesAnalysis,
-}: InfraBackendLibs) => {
+export const initGetLogEntryCategoriesRoute = ({ framework }: InfraBackendLibs) => {
   framework.registerRoute(
     {
       method: 'post',
       path: LOG_ANALYSIS_GET_LOG_ENTRY_CATEGORIES_PATH,
       validate: {
-        // short-circuit forced @kbn/config-schema validation so we can do io-ts validation
-        body: anyObject,
+        body: createValidationFunction(getLogEntryCategoriesRequestPayloadRT),
       },
     },
-    async (requestContext, request, response) => {
+    framework.router.handleLegacyErrors(async (requestContext, request, response) => {
```

Comment:
What does this handleLegacyErrors function do and where is it documented?

Reply:
I don't know of any documentation - I read the `http` service source. Inside of our handlers we use some platform services, which throw `Boom` exceptions. I found this helper improves the representation of these errors in the response. Without it we would have to capture and convert them ourselves to avoid masquerading them as generic code 500 errors.

Reply:
Actually, there is some here:

https://github.com/elastic/kibana/blob/2685654cdc67f6630a6c96feee09967c1f78034f/docs/development/core/server/kibana-plugin-core-server.irouter.handlelegacyerrors.md

Reply:
Thanks!